### PR TITLE
[Bugfix] Fix AbstractSliderChild crash with js-toolkit >= 3.4.3

### PR DESCRIPTION
**Pull request by antoine4livre**
### 🔗 Linked issue

#505 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

#### Problem

`AbstractSliderChild.mounted()` calls `this.$parent.$on("index", this)` which throws `TypeError: Cannot read properties of null (reading '$on')` with `@studiometa/js-toolkit` >= 3.4.3.

In js-toolkit 3.4.3, `$parent` changed from a property assigned during mounting to a dynamic getter that looks up registered instances at runtime. This getter can return `null` when a child's `mounted()` runs before the parent Slider is fully resolved in the instance registry.

Any page using `SliderBtn`, `SliderCount`, `SliderDots`, or `SliderProgress` crashes on mount.

#### Changes

**Commit 1 — Fix Slider child components**
- Replace `this.$parent` with a `slider` getter using `this.$closest('Slider')` in `AbstractSliderChild` and all subclasses (`SliderBtn`, `SliderDots`, `SliderProgress`)
- Defer `$on` subscription in `mounted()` with `nextFrame` to ensure the parent Slider is registered before the child looks it up
- Add null guards throughout
- Bump `@studiometa/js-toolkit` to `^3.5.0` (required for `$closest` support)

**Commit 2 — Fix AccordionItem**
- Replace `this.$parent` with `this.$closest('Accordion')` in `AccordionItem.mounted()`, consistent with the js-toolkit 3.5.0 deprecation of `$parent`

#### Out of scope

- **`Frame.ts`**: `this.$root.$update()` — deprecated but does not crash. Replacing with `this.$update()` would change behavior (only updates the Frame subtree instead of the entire app). Needs a separate discussion.
- **`$children`** (22 usages across 11 files) — deprecated in favor of `$query()` but still functional. Migration should be handled in a dedicated ticket.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have added tests (if possible).
- [x] I have updated the documentation accordingly.
- [x] I have updated the changelog.

### CODE REVIEW

File: packages/ui/Slider/SliderBtn.ts [hunk 1]
```diff
@@ -31,21 +31,23 @@ export class SliderBtn<T extends BaseProps = BaseProps> extends AbstractSliderCh
    * Update attributes.
    */
   update(index: number) {
-    if (this.$options.contain && !this.$parent.$options.contain) {
+    if (!this.slider) return;
```
Comment (titouanmathis) [inline packages/ui/Slider/SliderBtn.ts:34]: Prefer destructuring slider as a constant to avoid repeated lookups.

**Review by titouanmathis — CHANGES_REQUESTED** [review]

**Comment by titouanmathis** [comment]
Superseded by #507, which has been merged into `main` — the `AbstractSliderChild` crash (and the matching `AccordionItem` case) is now fixed there, so this branch no longer applies cleanly.

Thanks @antoine4livre — your diagnosis was spot on (the `$parent` dynamic getter returning `null` when a child mounts before its parent). #507 landed on the same `$parent` → `$closest` direction, and additionally routes the current index through a per-instance store with a parent-side handshake so the connection is deterministic regardless of mount order, targeting `@studiometa/js-toolkit@^3.6.0` (which corrects `$closest` and adds `createStorage`). It also carries regression tests for the mount-ordering and `contain`-mode edge cases.

Closing this one as the fix is now on `main`. 🙏
